react/src/components/ServiceLauncherPageContent.tsx

```diff
@@ -1017,15 +1149,61 @@ const ServiceLauncherPageContent: React.FC<ServiceLauncherPageContentProps> = ({
                         >
                           <ResourceAllocationFormItems enableResourcePresets />
                         </div>
-                        <Form.Item
-                          label={t('session.launcher.EnvironmentVariable')}
-                        >
-                          <EnvVarFormList
-                            name={'envvars'}
-                            formItemProps={{
-                              validateTrigger: ['onChange', 'onBlur'],
-                            }}
-                          />
+                        <Form.Item dependencies={['runtimeVariant']} noStyle>
+                          {({ getFieldValue }) => {
+                            const runtimeVariant =
+                              getFieldValue('runtimeVariant');
+                            const runtimeVariantConfig = runtimeVariant
+                              ? RUNTIME_ENV_VAR_CONFIGS[runtimeVariant]
+                              : null;
+
+                            return (
+                              <Form.Item
+                                label={t(
+                                  'session.launcher.EnvironmentVariable',
+                                )}
+                              >
+                                <EnvVarFormList
+                                  name={'envvars'}
+                                  requiredEnvVars={
+                                    runtimeVariantConfig?.requiredEnvVars
+                                  }
+                                  optionalEnvVars={
+                                    runtimeVariantConfig?.optionalEnvVars
+                                  }
+                                  disableSanitizeKeys={['NGC_API_KEY']}
+                                  formItemProps={{
+                                    validateTrigger: ['onChange', 'onBlur'],
+                                    rules: [
+                                      {
+                                        warningOnly: true,
+                                        validator: async (
+                                          _rule,
+                                          value: string,
+                                        ) => {
+                                          if (!value) {
+                                            return Promise.resolve();
+                                          }
+
+                                          const validateVariable =
+                                            createValidateVariableFunction(
+                                              runtimeVariant,
+                                            );
+                                          if (!validateVariable(value)) {
```

Comment:
Since you are not creating a function to reuse multiple times, making it a function that directly returns the result is better for readability.
```suggestion
                                          if (!validateVariable(runtimeVariant, value)) {
```